### PR TITLE
[CherryPick] Ray core doc changes #32153, #32191, #32331

### PR DESCRIPTION
**Pull request by jjyao**
## Why are these changes needed?
#32153, #32191, #32331

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by cadedaniel** [comment]
Let's wait on https://github.com/ray-project/ray/pull/32446 before merging, that is on the critical path for release wheel

**Review by cadedaniel — APPROVED** [review]
Ready for approval! Not blocking the release tests today.

**Review by zhe-thoughts — APPROVED** [review]
Approved for picking into 2.3.0

**Pull request merged**
